src/components/button-swap/button-swap.test.ts

```diff
@@ -0,0 +1,177 @@
+    expect(statusMsgs.length).toBe(1);
+
+    const firstBtn = await screen.getByRole('button', {
+      name: /^get notifications$/i,
+    });
+    await expect(firstBtn).toBeVisible();
+
+    const secondBtn = await screen.queryByRole('button', {
+      name: /^turn off notifications$/i,
+    });
+    expect(secondBtn).toBeNull();
+  })
+);
+
+test(
+  'Swap UI state when clicked',
+  withBrowser(async ({ utils, screen, user }) => {
+    await utils.injectHTML(await componentMarkup());
+    await loadGlobalCSS(utils);
+    // I'd like to avoid using a test ID, but I couldn't figure out a different way.
+    // @todo Can this be done without at test ID?
+    await initJS(utils);
+
+    let firstBtn = await screen.queryByRole('button', {
+      name: /^get notifications$/i,
+    });
+    await expect(firstBtn).not.toHaveClass('is-slashed');
+
+    // Button swap action
+    await user.click(firstBtn);
+
+    // Visually hidden text for a more inclusive UX
+    let statusMsg = await screen.getByRole('alert');
+    await expect(statusMsg).toHaveTextContent(
+      /^currently subscribed to notifications$/i
+    );
+    await expect(statusMsg).toHaveFocus();
+
+    // Query for first button again in its new state
+    firstBtn = await screen.queryByRole('button', {
+      name: /^get notifications$/i,
+    });
+    expect(firstBtn).toBeNull();
+
+    let secondBtn = await screen.queryByRole('button', {
+      name: /^turn off notifications$/i,
+    });
+    await expect(secondBtn).toBeVisible();
+    await expect(secondBtn).toHaveClass('is-slashed');
+
+    // Button swap action
+    await user.click(secondBtn);
+
+    // Visually hidden text for a more inclusive UX
+    statusMsg = await screen.getByRole('alert');
+    await expect(statusMsg).toHaveTextContent(
+      /^currently unsubscribed from notifications$/i
+    );
+    await expect(statusMsg).toHaveFocus();
+
+    // Query for second button again in its new state
+    secondBtn = await screen.queryByRole('button', {
+      name: /^turn off notifications$/i,
+    });
+    expect(secondBtn).toBeNull();
+
+    // Query for first button again in its new state
+    firstBtn = await screen.queryByRole('button', {
+      name: /^get notifications$/i,
+    });
+    await expect(firstBtn).toBeVisible();
+    await expect(firstBtn).not.toHaveClass('is-slashed');
+  })
+);
+
+test(
+  'Set custom messages and labels',
+  withBrowser(async ({ utils, screen, user }) => {
+    await utils.injectHTML(
+      await componentMarkup({
+        initial_visual_label: 'Hello world',
+        swapped_visual_label: 'Have a great day',
+        initial_label: 'Unsubscribed',
+        swapped_label: 'Subscribed',
+      })
+    );
+    await loadGlobalCSS(utils);
+    await initJS(utils);
+
+    // Visually hidden text for a more inclusive UX
+    let statusMsg = await screen.getByRole('status');
+    await expect(statusMsg).toHaveTextContent(/^unsubscribed$/i);
```

Comment:
Nit: It might make sense to use a string directly instead of a regex here since the static value is written directly in this test

Reply:
Done! https://github.com/cloudfour/cloudfour.com-patterns/pull/1516/commits/dc3cca71b226578e8799fb575d1115a646a5fe87